aten/src/ATen/core/TensorMethods.h

```diff
@@ -78,8 +79,13 @@ inline Tensor & Tensor::names_(c10::optional<DimnameList> names) const {
 #ifdef USE_STATIC_DISPATCH
     return TypeDefault::names_(const_cast<Tensor&>(*this), names);
 #else
-    static auto table = globalATenDispatch().getOpTable("aten::names_(Tensor(a!) self, Dimname[]? names) -> Tensor(a!)");
-    return table->getOp<Tensor & (Tensor &, c10::optional<DimnameList>)>(type_set(), is_variable())(const_cast<Tensor&>(*this), names);
+    static c10::OperatorHandle op = c10::Dispatcher::singleton().findSchema({"aten::names_", ""}).value();
+    if (is_variable()) {
```

Comment:
why is variable switch lives in each and every method? can't it live inside dispatcher? it'd make Ed's changes to multidispatch easier

Reply:
this is now gone and uses VariableTensorId